frame/mosaic/src/lib.rs

```diff
@@ -0,0 +1,676 @@
+			keep_alive: bool,
+		) -> DispatchResultWithPostInfo {
+			let caller = ensure_signed(origin)?;
+			ensure!(AssetsInfo::<T>::contains_key(asset_id), Error::<T>::UnsupportedAsset);
+			let network_info =
+				NetworkInfos::<T>::get(network_id.clone()).ok_or(Error::<T>::UnsupportedNetwork)?;
+			ensure!(network_info.enabled, Error::<T>::NetworkDisabled);
+			ensure!(network_info.max_transfer_size >= amount, Error::<T>::ExceedsMaxTransferSize);
+
+			T::Assets::transfer(
+				asset_id,
+				&caller,
+				&Self::sub_account_id(SubAccount::outgoing(caller.clone())),
+				amount,
+				keep_alive,
+			)?;
+			let now = <frame_system::Pallet<T>>::block_number();
+			let lock_until = now.safe_add(&TimeLockPeriod::<T>::get())?;
+
+			OutgoingTransactions::<T>::try_mutate(
+				caller.clone(),
+				asset_id,
+				|prev| -> Result<(), DispatchError> {
+					match prev.as_mut() {
+						// If we already have an outgoing tx, we update the lock_time and add the
+						// amount.
+						Some((already_locked, _)) => {
+							let amount = amount.safe_add(already_locked)?;
+							*prev = Some((amount, lock_until))
+						},
+						None => *prev = Some((amount, lock_until)),
+					}
+					Ok(())
+				},
+			)?;
+
+			let id = generate_id::<T>(&caller, &network_id, &asset_id, &address, &amount, &now);
+			Self::deposit_event(Event::<T>::TransferOut { to: address, amount, network_id, id });
+
+			Ok(().into())
+		}
+
+		/// Called by the relayer to confirm that it will relay a transaction, disabling the user
+		/// from reclaiming their tokens.
+		///
+		/// # Restrictions
+		/// - Origin must be relayer
+		/// - Outgoing transaction must exist for the user
+		/// - Amount must be equal or lower than what the user has locked
+		///
+		/// # Note
+		/// - Reclaim period is not reset if not all the funds are moved; menaing that the clock
+		///   remains ticking for the relayer to pick up the rest of the transaction.
+		#[pallet::weight(10_000)]
+		#[transactional]
+		pub fn accept_transfer(
+			origin: OriginFor<T>,
+			from: AccountIdOf<T>,
+			asset_id: AssetIdOf<T>,
+			amount: BalanceOf<T>,
+		) -> DispatchResultWithPostInfo {
+			ensure_relayer::<T>(origin)?;
+			OutgoingTransactions::<T>::try_mutate_exists::<_, _, _, DispatchError, _>(
+				from.clone(),
+				asset_id,
+				|maybe_tx| match *maybe_tx {
+					Some((balance, lock_period)) => {
+						ensure!(amount <= balance, Error::<T>::AmountMismatch);
+						T::Assets::burn_from(
+							asset_id,
+							&Self::sub_account_id(SubAccount::outgoing(from.clone())),
+							amount,
+						)?;
+
+						// No remaing funds need to be transferred for this asset, so we can delete
+						// the storage item.
+						if amount == balance {
+							*maybe_tx = None
+						} else {
+							let new_balance =
+								balance.checked_sub(&amount).ok_or(Error::<T>::AmountMismatch)?;
+							*maybe_tx = Some((new_balance, lock_period));
+						}
+
+						Self::deposit_event(Event::<T>::TransferAccepted {
```

Comment:
accepted transfer can be partially accepted transfer

Reply:
I've reflected this in the events.